### PR TITLE
CAMEL-10401: Support chmod for ftp/ftps

### PR DESCRIPTION
**Pull request by eidottermihi**

### CODE REVIEW

File: components/camel-ftp/src/test/java/org/apache/camel/component/file/remote/FromFileToFtpWithChmodTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * der Landeshauptstadt München, 2016
+ */
+package org.apache.camel.component.file.remote;
+
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.junit.Test;
+
+/**
+ * @author eidottermihi
```
Comment (davsclaus) [inline components/camel-ftp/src/test/java/org/apache/camel/component/file/remote/FromFileToFtpWithChmodTest.java]: We do not have author tags in the source code. Do you mind removing this?

File: components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpOperations.java [hunk 2]
```diff
@@ -593,7 +593,17 @@ private boolean doStoreFile(String name, String targetName, Exchange exchange) t
...
+            // after storing file, we may set chmod on the file
+            String chmod = ((FtpConfiguration) endpoint.getConfiguration()).getChmod();
+            if (ObjectHelper.isNotEmpty(chmod)) {
+                log.debug("Setting chmod: {} on file: {}", chmod, targetName);
+                String command = "chmod " + chmod + " " + targetName;
+                log.trace("Client sendSiteCommand: {}", command);
+                boolean success = client.sendSiteCommand(command);
+                log.trace("Client sendSiteCommand successful: {}", success);
+            }
+
```
Comment (davsclaus) [inline components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpOperations.java]: I think we need to store the client reply code / string before doing chmod operation, so they are from the operation that performed the upload. So eg move the above code below the code below that set those headers.

File: components/camel-ftp/src/test/java/org/apache/camel/component/file/remote/FromFileToFtpWithChmodTest.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/*
```
Comment (davsclaus) [inline components/camel-ftp/src/test/java/org/apache/camel/component/file/remote/FromFileToFtpWithChmodTest.java]: The license header in the top of the file should be ASF standard license we use, so you can copy it from another file.

**Comment by eidottermihi** [comment]
@davsclaus Thanks for the review. I've changed them accordingly.

**Comment by davsclaus** [comment]
Thanks for the PR it has been merged. Do you mind closing this?

**Pull request closed without merging**